src/aind_data_schema_models/modalities.py

```diff
@@ -1,144 +1,33 @@
-
-
-class Confocal(_Modality):
-    """Confocal"""
-
-    name: Literal["Confocal microscopy"] = "Confocal microscopy"
-    abbreviation: Literal["confocal"] = "confocal"
-
-
-class Ecephys(_Modality):
-    """Ecephys"""
-
-    name: Literal["Extracellular electrophysiology"] = "Extracellular electrophysiology"
-    abbreviation: Literal["ecephys"] = "ecephys"
-
-
-class Electromyography(_Modality):
-    """Electromyography"""
-
-    name: Literal["Electromyography"] = "Electromyography"
-    abbreviation: Literal["EMG"] = "EMG"
-
-
-class Fmost(_Modality):
-    """Fmost"""
-
-    name: Literal["Fluorescence micro-optical sectioning tomography"] = (
-        "Fluorescence micro-optical sectioning tomography"
-    )
-    abbreviation: Literal["fMOST"] = "fMOST"
-
-
-class Icephys(_Modality):
-    """Icephys"""
-
-    name: Literal["Intracellular electrophysiology"] = "Intracellular electrophysiology"
-    abbreviation: Literal["icephys"] = "icephys"
-
-
-class Isi(_Modality):
-    """Intrinsic signal imaging"""
-
-    name: Literal["Intrinsic signal imaging"] = "Intrinsic signal imaging"
-    abbreviation: Literal["ISI"] = "ISI"
-
-
-class Fib(_Modality):
-    """Fib"""
-
-    name: Literal["Fiber photometry"] = "Fiber photometry"
-    abbreviation: Literal["fib"] = "fib"
-
-
-class Merfish(_Modality):
-    """Merfish"""
-
-    name: Literal["Multiplexed error-robust fluorescence in situ hybridization"] = (
-        "Multiplexed error-robust fluorescence in situ hybridization"
-    )
-    abbreviation: Literal["merfish"] = "merfish"
-
-
-class Mri(_Modality):
-    """Mri"""
-
-    name: Literal["Magnetic resonance imaging"] = "Magnetic resonance imaging"
-    abbreviation: Literal["MRI"] = "MRI"
-
-
-class POphys(_Modality):
-    """POphys"""
-
-    name: Literal["Planar optical physiology"] = "Planar optical physiology"
-    abbreviation: Literal["ophys"] = "ophys"
-
-
-class Slap(_Modality):
-    """Slap"""
-
-    name: Literal["Scanned line projection imaging"] = "Scanned line projection imaging"
-    abbreviation: Literal["slap"] = "slap"
-
-
-class Spim(_Modality):
-    """Spim"""
-
-    name: Literal["Selective plane illumination microscopy"] = "Selective plane illumination microscopy"
-    abbreviation: Literal["SPIM"] = "SPIM"
-
+Modalities = create_literal_class(
+    objects=read_csv("models/modalities.csv"),
+    class_name="Modalities",
+    base_model=Modality,
+    discriminator="abbreviation",
+    class_module=__name__,
+)
 
-class Modality:
-    """Modality classes"""
 
-    BEHAVIOR = Behavior()
-    BEHAVIOR_VIDEOS = BehaviorVideos()
-    CONFOCAL = Confocal()
-    ECEPHYS = Ecephys()
-    EMG = Electromyography()
-    FMOST = Fmost()
-    ICEPHYS = Icephys()
-    FIB = Fib()
-    ISI = Isi()
-    MERFISH = Merfish()
-    MRI = Mri()
-    POPHYS = POphys()
-    SLAP = Slap()
-    SPIM = Spim()
-    _ALL = tuple(_Modality.__subclasses__())
-    ONE_OF = Annotated[Union[_ALL], Field(discriminator="name")]
+@classmethod
```

Comment:
You can remove the decorator and consider making the method private, or putting it into the utils package

Reply:
When I removed the decorator it stopped passing `self` correctly, so I think I still need it.